Missing ARCH, CROSS_COMPILE environment variable

It would be nice to set the `ARCH` environment variable in each image. It looks redundant to set while running the container, i.e. `docker run ... -e ARCH=arm dockercross/linux-armv7`

Maybe I'm missing the point how to use these images, I needed to set `-e CROSS_COMPILE=` also in order to compile. What I'm missing here?

Common steps:
```
~ docker run --rm dockcross/linux-armv7 > dockcross-linux-armv7
~ chmod +x dockcross-linux-armv7
```

Failed approach:
```
./dockcross-linux-armv7 -a "-e ARCH=arm" bash -c "make tegra3_android_defconfig && make -j 4"
*** Default configuration is based on 'versatile_defconfig'
drivers/misc/inv_mpu6050/Kconfig:22:warning: choice value used outside its choice group
drivers/misc/inv_mpu6050/Kconfig:31:warning: choice value used outside its choice group
drivers/misc/inv_mpu6050/Kconfig:40:warning: choice value used outside its choice group
drivers/misc/inv_mpu6050/Kconfig:56:warning: choice value used outside its choice group
drivers/misc/inv_mpu6050/Kconfig:60:warning: choice value used outside its choice group
#
# configuration written to .config
#
scripts/kconfig/conf --silentoldconfig Kconfig
drivers/misc/inv_mpu6050/Kconfig:22:warning: choice value used outside its choice group
drivers/misc/inv_mpu6050/Kconfig:31:warning: choice value used outside its choice group
drivers/misc/inv_mpu6050/Kconfig:40:warning: choice value used outside its choice group
drivers/misc/inv_mpu6050/Kconfig:56:warning: choice value used outside its choice group
drivers/misc/inv_mpu6050/Kconfig:60:warning: choice value used outside its choice group
  CHK     include/linux/version.h
  CHK     include/generated/utsrelease.h
  CC      scripts/mod/empty.o
gcc: error: unrecognized argument in option '-mabi=apcs-gnu'
gcc: note: valid arguments to '-mabi=' are: ms sysv
gcc: error: unrecognized command line option '-mlittle-endian'
gcc: error: unrecognized command line option '-mapcs'
gcc: error: unrecognized command line option '-mno-sched-prolog'
scripts/Makefile.build:305: recipe for target 'scripts/mod/empty.o' failed
make[2]: *** [scripts/mod/empty.o] Error 1
scripts/Makefile.build:441: recipe for target 'scripts/mod' failed
make[1]: *** [scripts/mod] Error 2
Makefile:506: recipe for target 'scripts' failed
make: *** [scripts] Error 2
make: *** Waiting for unfinished jobs....
```

Successful approach:
```
./dockcross-linux-armv7 -a "-e ARCH=arm -e CROSS_COMPILE=\"arm-linux-gnueabihf-\"" bash -c "make tegra3_android_defconfig && make -j 4"
```

--- a/linux-arm64/Dockerfile.in
+++ b/linux-arm64/Dockerfile.in
@@ -31,6 +31,11 @@ ENV CMAKE_TOOLCHAIN_FILE ${CROSS_ROOT}/Toolchain.cmake
 
 ENV PKG_CONFIG_PATH /usr/lib/aarch64-linux-gnu/pkgconfig
 
+# Linux kernel cross compilation variables
+ENV PATH ${PATH}:${CROSS_ROOT}/bin
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH arm64
+
 # Build-time metadata as defined at http://label-schema.org
 ARG BUILD_DATE
 ARG IMAGE=dockcross/linux-arm64

--- a/linux-armv5-musl/Dockerfile.in
+++ b/linux-armv5-musl/Dockerfile.in
@@ -33,6 +33,11 @@ ENV CMAKE_TOOLCHAIN_FILE ${CROSS_ROOT}/Toolchain.cmake
 
 ENV PKG_CONFIG_PATH /usr/lib/arm-linux-musleabihf/
 
+# Linux kernel cross compilation variables
+ENV PATH ${PATH}:${CROSS_ROOT}/bin
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH arm
+
 # Build-time metadata as defined at http://label-schema.org
 ARG BUILD_DATE
 ARG IMAGE=dockcross/linux-armv5-musl

--- a/linux-armv5/Dockerfile.in
+++ b/linux-armv5/Dockerfile.in
@@ -35,6 +35,11 @@ ENV CMAKE_TOOLCHAIN_FILE ${CROSS_ROOT}/Toolchain.cmake
 
 ENV PKG_CONFIG_PATH /usr/lib/arm-linux-gnueabihf/
 
+# Linux kernel cross compilation variables
+ENV PATH ${PATH}:${CROSS_ROOT}/bin
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH arm
+
 # Build-time metadata as defined at http://label-schema.org
 ARG BUILD_DATE
 ARG IMAGE=dockcross/linux-armv5

--- a/linux-armv6/Dockerfile
+++ b/linux-armv6/Dockerfile
@@ -43,6 +43,11 @@ ENV QEMU_SET_ENV "LD_LIBRARY_PATH=${CROSS_ROOT}/lib:${CROSS_ROOT}/libc/lib/${CRO
 COPY Toolchain.cmake ${CROSS_ROOT}/
 ENV CMAKE_TOOLCHAIN_FILE ${CROSS_ROOT}/Toolchain.cmake
 
+# Linux kernel cross compilation variables
+ENV PATH ${PATH}:${CROSS_ROOT}/bin
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH arm
+
 # Build-time metadata as defined at http://label-schema.org
 ARG BUILD_DATE
 ARG IMAGE=dockcross/linux-armv6

--- a/linux-armv7/Dockerfile.in
+++ b/linux-armv7/Dockerfile.in
@@ -31,6 +31,11 @@ ENV CMAKE_TOOLCHAIN_FILE ${CROSS_ROOT}/Toolchain.cmake
 
 ENV PKG_CONFIG_PATH /usr/lib/arm-linux-gnueabihf/
 
+# Linux kernel cross compilation variables
+ENV PATH ${PATH}:${CROSS_ROOT}/bin
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH arm
+
 # Build-time metadata as defined at http://label-schema.org
 ARG BUILD_DATE
 ARG IMAGE=dockcross/linux-armv7

--- a/linux-armv7a/Dockerfile.in
+++ b/linux-armv7a/Dockerfile.in
@@ -34,6 +34,11 @@ ENV CMAKE_TOOLCHAIN_FILE ${CROSS_ROOT}/Toolchain.cmake
 
 ENV PKG_CONFIG_PATH /usr/lib/arm-linux-gnueabihf/
 
+# Linux kernel cross compilation variables
+ENV PATH ${PATH}:${CROSS_ROOT}/bin
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH arm
+
 # Build-time metadata as defined at http://label-schema.org
 ARG BUILD_DATE
 ARG IMAGE=dockcross/linux-armv7a

--- a/linux-mips/Dockerfile.in
+++ b/linux-mips/Dockerfile.in
@@ -29,6 +29,11 @@ ENV QEMU_SET_ENV "LD_LIBRARY_PATH=${CROSS_ROOT}/lib:${QEMU_LD_PREFIX}"
 COPY Toolchain.cmake ${CROSS_ROOT}/
 ENV CMAKE_TOOLCHAIN_FILE ${CROSS_ROOT}/Toolchain.cmake
 
+# Linux kernel cross compilation variables
+ENV PATH ${PATH}:${CROSS_ROOT}/bin
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH mips
+
 # Build-time metadata as defined at http://label-schema.org
 ARG BUILD_DATE
 ARG IMAGE=dockcross/linux-mips

--- a/linux-mipsel/Dockerfile
+++ b/linux-mipsel/Dockerfile
@@ -32,6 +32,11 @@ ENV QEMU_SET_ENV "LD_LIBRARY_PATH=${CROSS_ROOT}/lib:${CROSS_ROOT}/libc/lib/${CRO
 COPY Toolchain.cmake ${CROSS_ROOT}/
 ENV CMAKE_TOOLCHAIN_FILE ${CROSS_ROOT}/Toolchain.cmake
 
+# Linux kernel cross compilation variables
+ENV PATH ${PATH}:${CROSS_ROOT}/bin
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH mips
+
 # Build-time metadata as defined at http://label-schema.org
 ARG BUILD_DATE
 ARG IMAGE=dockcross/linux-mipsel

--- a/linux-ppc64le/Dockerfile
+++ b/linux-ppc64le/Dockerfile
@@ -40,6 +40,11 @@ WORKDIR /work
 COPY Toolchain.cmake /usr/lib/${CROSS_TRIPLE}/
 ENV CMAKE_TOOLCHAIN_FILE /usr/lib/${CROSS_TRIPLE}/Toolchain.cmake
 
+# Linux kernel cross compilation variables
+ENV PATH ${PATH}:${CROSS_ROOT}/bin
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH powerpc
+
 # Build-time metadata as defined at http://label-schema.org
 ARG BUILD_DATE
 ARG IMAGE=dockcross/linux-ppc64le

--- a/linux-s390x/Dockerfile.in
+++ b/linux-s390x/Dockerfile.in
@@ -29,6 +29,11 @@ ENV QEMU_SET_ENV "LD_LIBRARY_PATH=${CROSS_ROOT}/lib:${QEMU_LD_PREFIX}"
 COPY Toolchain.cmake ${CROSS_ROOT}/
 ENV CMAKE_TOOLCHAIN_FILE ${CROSS_ROOT}/Toolchain.cmake
 
+# Linux kernel cross compilation variables
+ENV PATH ${PATH}:${CROSS_ROOT}/bin
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH s390
+
 # Build-time metadata as defined at http://label-schema.org
 ARG BUILD_DATE
 ARG IMAGE=dockcross/linux-s390x

--- a/linux-x86/Dockerfile
+++ b/linux-x86/Dockerfile
@@ -36,6 +36,10 @@ ENV AS=${CROSS_ROOT}/bin/${CROSS_TRIPLE}-as \
 COPY Toolchain.cmake /usr/lib/${CROSS_TRIPLE}/
 ENV CMAKE_TOOLCHAIN_FILE /usr/lib/${CROSS_TRIPLE}/Toolchain.cmake
 
+# Linux kernel cross compilation variables
+ENV CROSS_COMPILE ${CROSS_TRIPLE}-
+ENV ARCH x86
+
 COPY linux32-entrypoint.sh /dockcross/
 ENTRYPOINT ["/dockcross/linux32-entrypoint.sh"]
 

